installers/msi-language/ActiveState/Error.cs

```diff
@@ -35,35 +34,57 @@ public static void ResetErrorDetails(Session session)
         }
 
         /// <summary>
-        /// SetErrorDetails writes the network error details to the user's registry.
+        /// SetDetails writes the error details to the user's registry.
         /// This function must be run from a deferred custom action
         /// </summary>
-        public static void SetErrorDetails(Session session, string msg)
-        {
-            SetErrorDetails(
-                session,
-                string.Format("HKEY_USERS\\{0}\\SOFTWARE\\ActiveState\\{1}", session.CustomActionData["USERSID"], session.CustomActionData["PRODUCT_NAME"]),
-                msg
-            );
-        }
-
-        /// <summary>
-        /// SetErrorDetails writes the network error details to the user's registry
-        /// </summary>
-        private static void SetErrorDetails(Session session, string registryKey, string msg)
+        public static void SetDetails(Session session, string errorType, string msg)
         {
+            string registryKey = string.Format("HKEY_USERS\\{0}\\SOFTWARE\\ActiveState\\{1}", session.CustomActionData["USERSID"], session.CustomActionData["PRODUCT_NAME"]);
             RegistryValueKind registryEntryDataType = RegistryValueKind.String;
+            session.Log("Setting registry key: {0}", registryKey);
             try
             {
-                Registry.SetValue(registryKey, networkErrorKey, "true", registryEntryDataType);
-                Registry.SetValue(registryKey, networkErrorMessageKey, msg, registryEntryDataType);
+                Registry.SetValue(registryKey, TypeRegistryKey, errorType, registryEntryDataType);
+                Registry.SetValue(registryKey, MessageRegistryKey, msg, registryEntryDataType);
             }
             catch (Exception registryException)
             {
-                string registryExceptionMsg = string.Format("Could not set network error registry values. Exception: {0}", registryException.ToString());
+                string registryExceptionMsg = string.Format("Could not set error registry values. Exception: {0}", registryException.ToString());
                 session.Log(registryExceptionMsg);
                 RollbarReport.Error(registryExceptionMsg, session);
             }
         }
     }
+
+    public class SecurityError
+    {
+        private const string type = "Antivirus";
+
+        public static void SetDetails(Session session, string msg)
+        {
+            session.Log("Setting security error details");
+            Error.SetDetails(session, type, msg);
+        }
+
+        public static string Type()
+        {
+            return type;
+        }
+    }
+
+    public class NetworkError
```

Comment:
These 2 are basically identical aside from type. I'm sure C# has class inheritance that you could use here?

Reply:
Yes we can. It changes how we use the errors but removes the duplicated code.